core/sched.c

```diff
@@ -80,11 +80,14 @@ int __attribute__((used)) sched_run(void)
 {
     sched_context_switch_request = 0;
 
+#ifndef MODULE_CORE_IDLE_THREAD
+    while (!runqueue_bitcache) {
```

Comment:
note to self: as is, the compiler must assume that sched_arch_idle() calls a function that modifies runqueue_bitcache (it is static in this compilation unit). With LTO, it might come to a different result. need to make this atomic or volatile.